sdk/core/performance-stress/inc/azure/performance-stress/test_metadata.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Define the metadata of a performance test.
+ *
+ */
+
+#pragma once
+
+#include "azure/performance-stress/test.hpp"
+#include "azure/performance-stress/test_options.hpp"
+
+#include <functional>
+#include <memory>
+#include <string>
+
+namespace Azure { namespace PerformanceStress {
+  /**
+   * @brief Define the metadata of a test that can be run by the performance framework.
+   *
+   */
+  struct TestMetadata
```

Comment:
I like this design for encapsolating the Test.  However I would stick with a hash map of tests.  Most of the uses need to do a lookup and looking up with a unique name is more efficient than iterating over a list doing string comparisons.

To make the code easy to use we can define a type for it
```
using TestCollection = std::map<
        std::string,
        std::function<std::unique_ptr<Azure::PerformanceStress::PerformanceTest>(
            Azure::PerformanceStress::TestOptions)>>;
```
Then everywhere we use it we can do:
```
inline std::unique_ptr<Azure::PerformanceStress::PerformanceTest> PrintAvailableTests(
    TestCollection const& tests)
```

Reply:
You could go a step further and make the collection a class with a private member hash.  Then add helpers over the top to get/set tests into it.

Reply:
>  Most of the uses need to do a lookup and looking up with a unique name

We only need one search per application-run, and the number of the tests inside the application will never be that big (I would be impressed if it ever get to 100).  The improvement of using a Set for the O(1)  vs/ the O(n) in the Vector won't make a significant difference.

Also, looking for the test is not something impacting running the actual test.